zerver/lib/push_notifications.py

```diff
@@ -1188,3 +1188,55 @@ def handle_push_notification(user_profile_id: int, missed_message: Dict[str, Any
     user_identity = UserPushIdentityCompat(user_id=user_profile.id)
     send_apple_push_notification(user_identity, apple_devices, apns_payload)
     send_android_push_notification(user_identity, android_devices, gcm_payload, gcm_options)
+
+
+def send_test_push_notification_directly_to_device(
+    user_identity: UserPushIdentityCompat,
+    device: DeviceToken,
+    base_payload: Dict[str, Any],
+    remote: Optional["RemoteZulipServer"] = None,
+) -> None:
+    payload = base_payload
+    payload["event"] = "test-by-device-token"
+
+    if device.kind == RemotePushDeviceToken.APNS:
+        realm_uri = base_payload["realm_uri"]
+        apns_data = {
+            "alert": {
+                "title": _("Test notification"),
+                "body": _("This is a test notification from {realm_uri}.").format(
+                    realm_uri=realm_uri
+                ),
+            },
+            "sound": "default",
+            "custom": {"zulip": payload},
+        }
+        send_apple_push_notification(user_identity, [device], apns_data, remote=remote)
+    else:
+        assert device.kind == RemotePushDeviceToken.GCM
+        payload["time"] = timezone_now().timestamp()
```

Comment:
I think this should be changed to be an `int`, for consistency with message notifications. How about `datetime_to_timestamp(timezone_now())`? Looking back at this GitHub thread, I think that was used in an earlier revision.

Reply:
Oh yeah, thanks for catching that, I missed to update this part of the PR after that revision